src/editors/_shared/hooks/DataTypes/AbstractDataType.js

```diff
@@ -1,160 +1,189 @@
-import React from "react";
-import _toString from "lodash/toString";
+import React, { useState, useEffect, useMemo, useCallback } from "react";
 import { TextField, Typography } from "@material-ui/core";
 import { MonacoCodeEditor } from "@mov-ai/mov-fe-lib-code-editor";
-import { DATA_TYPES } from "../../../../utils/Constants";
+
+const identity = a => a;
+
+/*
+ * Handles initial value and onChange for data types that use strings as input
+ * Although they might output real objects
+ */
+export function useTextEdit(props) {
+  const { rowData = {}, dataType, onChange = () => {} } = props;
+  const initialValue = useMemo(
+    () => dataType.inputParsing.unparse(rowData.value),
+    [dataType]
+  );
+  const placeholder = useMemo(() => dataType.unparse(dataType.default), [dataType]);
+  const [value, setValue] = useState(initialValue || placeholder);
+
+  const handleOnChange = useCallback((value) => {
+    onChange(dataType.inputParsing.parse(value));
+    setValue(value);
+  }, [onChange, setValue, dataType]);
+
+  useEffect(() => {
+    if (rowData.value !== null)
+      setValue(dataType.inputParsing.unparse(rowData.value));
+  }, [handleOnChange, rowData.value]);
+
+  return { ...props, onChange: handleOnChange, value };
+}
+
+function StringEdit(props) {
+  const { onChange, dataType, ...rest } = useTextEdit(props);
+
+  return (<TextField
+    type={dataType.inputType}
+    inputProps={{ "data-testid": "input_value" }}
+    fullWidth
+    onChange={evt => onChange(evt.target.value)}
+    { ...rest }
+  />);
+}
+
+function CodeEdit(props) {
+  const { isNew, disabled, dataType, ...rest } = useTextEdit(props);
+
+  return (<Typography
+    data-testid="section_data-type-code-editor"
+    component="div"
+    style={{ height: "100px", width: "100%" }}
```

Comment:
Can we move this into css?

Reply:
Yes, we can. It wasn't before, but we can.